### PR TITLE
implemented postgres-ha solution

### PR DESCRIPTION
**Pull request by trodge**

### CODE REVIEW

**Comment by trodge** [comment]
I believe I have addressed all feedback.

**Comment by AlexBulankou** [comment]
Thanks @trodge. I'll share some additional verification instructions offline.

File: config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    availabilityType: REGIONAL
+    ipConfiguration:
+      ipv4Enabled: true
+      requireSsl: true
+      authorizedNetworks:
+      - name: mysql-ha-solution-cidr
+        value: 192.10.10.10/32 # {"$ref":"#/definitions/io.k8s.cli.setters.external-ip-range"}
+    locationPreference:
+      zone: us-central1-c # {"$ref":"#/definitions/io.k8s.cli.setters.zone"}
+---
```
Comment (AlexBulankou) [inline config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml:32]: Please ensure the following properties on main SQL instance (see https://paste.googleplex.com/5396719468544000):
activationPolicy: "ALWAYS", diskAuthresize: true, diskSize:10, diskType: "PD_SSD", pricingPlan: "PER_USE", replicationType: "SYNCHRONOUS", tier: "db-custom-2-13312".
For backupConfiguration, please ensure, binaryLogEnabled: false, enabled: true, startTime: "20:55"
For databaseFlags: name: "autovacuum", value: "off"
for maintenanceWindow: day:"7", hour: "12", updateTrack: "stable"

File: config-connector/solutions/sql/kpt/postgres-ha/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+| instance-name     | postgres-ha-solution   | package-default | name of SQL instance           | 9     |
+| password-1        | ${PASSWORD_1?}         | PLACEHOLDER     | password of  user              | 1     |
+| password-2        | ${PASSWORD_2?}         | PLACEHOLDER     | password of  user              | 1     |
+| password-3        | ${PASSWORD_3?}         | PLACEHOLDER     | password of  user              | 1     |
+| region            | us-central1            | package-default | region of SQL instance         | 4     |
+| username-1        | ${USERNAME_1?}         | PLACEHOLDER     | name of  user                  | 1     |
+| username-2        | ${USERNAME_2?}         | PLACEHOLDER     | name of  user                  | 1     |
+| username-3        | ${USERNAME_3?}         | PLACEHOLDER     | name of  user                  | 1     |
+| zone              | us-central1-c          | package-default | zone of main instance          | 1     |
+| zone-backup-1     | us-central1-a          | package-default | zone of first backup instance  | 1     |
```
Comment (AlexBulankou) [inline config-connector/solutions/sql/kpt/postgres-ha/README.md:27]: These are not backup instances, they are replication instances. Can you please refer to all field names as `replication` not `backup`.

File: config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    availabilityType: ZONAL
+    ipConfiguration:
+      ipv4Enabled: true
+      requireSsl: false
+      authorizedNetworks:
+      - name: mysql-ha-solution-cidr
+        value: 192.10.10.10/32 # {"$ref":"#/definitions/io.k8s.cli.setters.external-ip-range"}
+    locationPreference:
+      zone: us-central1-a # {"$ref":"#/definitions/io.k8s.cli.setters.zone-backup-1"}
+---
```
Comment (AlexBulankou) [inline config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml:51]: For all 3 replica instance please insure:
1. they are called -replicas, not backup
2. masterInstanceRef set to master database
3. activationPolicy:"ALWAYS", crashSafeReplication:true, diskAutoresize:true, diskSize:10, diskType:"PD_HDD", pricingPlan: "PER_USE", replicationType: "SYNCHRONOUS", tier: "db-custom-2-13312"
For databaseFlags: name: "autovacuum", value: "off"
for maintenanceWindow: day:"7", hour: "22", updateTrack: "stable"

Comment (trodge) [reply config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml:84]: Changes made

File: config-connector/solutions/sql/kpt/postgres-ha/README.md (hunk 2 above)
Comment (trodge) [reply config-connector/solutions/sql/kpt/postgres-ha/README.md]: Done

File: config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml (hunk 1 above)
Comment (trodge) [reply config-connector/solutions/sql/kpt/postgres-ha/sqlinstance.yaml:49]: Done

**Pull request merged**